plat/common/include/x86/acpi/acpi.h

```diff
@@ -37,25 +38,23 @@
 #include <x86/acpi/sdt.h>
 #include <x86/acpi/madt.h>
 
-struct RSDPDescriptor {
-	char Signature[8];
-	__u8 Checksum;
-	char OEMID[6];
-	__u8 Revision;
-	__u32 RsdtAddress;
-} __packed;
+#define RSDP_SIG		"RSD PTR "
+#define RSDP_SIG_LEN		8
 
-struct RSDPDescriptor20 {
-	struct RSDPDescriptor v1;
-
-	__u32 Length;
-	__u64 XsdtAddress;
-	__u8 ExtendedChecksum;
-	__u8 Reserved[3];
-} __packed;
+typedef struct acpi_rsdp {
```

Comment:
Nope, just personal preference :). I thought that it would be misaligned with some other structures we already define, but I noticed that there are other `_t` typedefs as well, so I thought it would be fine. :confused:

Reply:
I think there are only a few occurrences for struct typedefs in the current code base. Most of them are in the (imported?) Xen platform.

Reply:
I have used tons of these in my [EFI stub PR](https://github.com/unikraft/unikraft/pull/909/commits/7a22d982562c3df6c9a18a8dbeadba97f6a31d9b) as well :cry:. I just like them because they keep declarations and casts shorter. @marcrittinghaus what is your opinion on using struct typedefs on Unikraft's codebase?